deepmd/dpmodel/atomic_model/linear_atomic_model.py

```diff
@@ -225,32 +225,30 @@ def fitting_output_def(self) -> FittingOutputDef:
             ]
         )
 
-    @staticmethod
-    def serialize(models, type_map) -> dict:
+    def serialize(self) -> dict:
         return {
             "@class": "Model",
             "type": "linear",
             "@version": 1,
-            "models": [model.serialize() for model in models],
-            "model_name": [model.__class__.__name__ for model in models],
-            "type_map": type_map,
+            "models": [model.serialize() for model in self.models],
+            "type_map": self.type_map,
         }
 
-    @staticmethod
-    def deserialize(data) -> Tuple[List[BaseAtomicModel], List[str]]:
+    @classmethod
+    def deserialize(cls, data: dict) -> "LinearEnergyAtomicModel":
         data = copy.deepcopy(data)
         check_version_compatibility(data.pop("@version", 1), 1, 1)
         data.pop("@class")
         data.pop("type")
-        model_names = data["model_name"]
-        type_map = data["type_map"]
+        type_map = data.pop("type_map")
         models = [
-            getattr(sys.modules[__name__], name).deserialize(model)
-            for name, model in zip(model_names, data["models"])
+            BaseAtomicModel.get_class_by_type(model["type"]).deserialize(model)
+            for model in data["models"]
         ]
-        return models, type_map
+        data.pop("models")
+        return cls(models, type_map, **data)
 
-    @abstractmethod
+    # @abstractmethod
```

Comment:
If we want to use the serialize/deserialize method from LinearAtomicModel in ZBLAtomicModel, we have to instantiate a concrete class, thus need to remove the `@abstractmethod`

Reply:
It might not be a good idea to instantiate it. We may need another method just to handle data, but doesn't instantiate the object.

Reply:
@njzjz I updated the PT side, see if that's what you have in mind.